sdks/python/apache_beam/transforms/core.py

```diff
@@ -320,6 +320,22 @@ def split_and_size(self, element, restriction):
     for part in self.split(element, restriction):
       yield part, self.restriction_size(element, part)
 
+  def truncate(self, element, restriction):
+    """Truncate the given restriction into finite amount of work when the
+    pipeline starts to drain.
+
+    By default, if the restriction is bounded, it will return the entire current
+    restriction. If the restriction is unbounded, it will return None.
+
+    The method throws NotImplementError when RestrictionTracker.is_bounded() is
+    not implemented.
+
+    It's recommended to implement this API if more granularity is required.
+    """
+    restriction_tracker = self.create_tracker(restriction)
+    if restriction_tracker.is_bounded():
+      return restriction
```

Comment:
```suggestion
      yield restriction
```

Reply:
I think we should return here since we expect `truncate` to produce only one truncated restriction per input restriction.

Reply:
We expect 0 or 1 which is why the yield is important. It also wouldn't matter if they returned more then 1. It might even help the drain complete sooner.